keps/sig-storage/596-csi-inline-volumes/README.md

```diff
@@ -213,32 +250,214 @@ Inline volume requests can only participate in mount/unmount volume operations.
 * Expected generated volumeHandle is created properly: https://github.com/kubernetes/kubernetes/blob/10005d2e1e1425904f8c7bf5615e730fb0fea7c9/pkg/volume/csi/csi_plugin_test.go#L177
-* Ensure that CSIDriver.Spec.Mode field is validated properly: https://github.com/kubernetes/kubernetes/pull/80568
+* Ensure that CSIDriver.VolumeLifecycleModes field is validated properly: https://github.com/kubernetes/kubernetes/pull/80568
 * Ensure volumeHandle conforms to resource naming format: TODO
 * CSIVolumeSource info persists in CSI json file during mount/unmount: TODO
-* Ensure Kubelet skips attach/detach when `CSIDriver.Mode = ephemeral`: TODO
-* Ensure Kubelet skips inline logic when `CSIDriver.Mode = persistent` or `CSIDriver.Mode is empty`: covered by existing tests
+* Ensure Kubelet skips attach/detach when `CSIDriver.VolumeLifecycleModes = Ephemeral`: TODO
+* Ensure Kubelet skips inline logic when `CSIDriver.VolumeLifecycleModes = Persistent` or `CSIDriver.VolumeLifecycleModes is empty`: covered by existing tests
+
+#### E2E tests
 
-### E2E tests
 * Pod spec with an ephemeral inline volume request can be mounted/unmounted: https://github.com/pohly/kubernetes/blob/4bc5d065c919fc239e2c8b40e6a96e409ca011fd/test/e2e/storage/csi_mock_volume.go#L356-L371, https://github.com/pohly/kubernetes/blob/4bc5d065c919fc239e2c8b40e6a96e409ca011fd/test/e2e/storage/testsuites/ephemeral.go#L110-L115
 * Two pods accessing an ephemeral inline volume which has the same attributes in both pods: "should support two pods which share the same data" in `ephemeral.go` (upcoming PR)
 * Single pod referencing two distinct inline volume request from the same driver: "should support multiple inline ephemeral volumes" in `ephemeral.go` (upcoming PR)
 * CSI Kubelet code invokes driver operations during mount for ephemeral volumes: `checkPodLogs` in `csi_mock_volume.go` (upcoming PR)
 * CSI Kubelet code invokes driver operation during unmount of ephemeral volumes: `checkPodLogs` in `csi_mock_volume.go` (upcoming PR)
 * CSI Kubelet cleans up ephemeral volume paths once pod goes away: TODO
-* Apply PodSecurity settings for allowed CSI drivers: TODO
 * Enable testing of an external ephemeral CSI driver: https://github.com/kubernetes/kubernetes/pull/79983/files#diff-e5fc8d9911130b421b74b1ebc273f458
 * Enable testing of the csi-host-path-driver in ephemeral mode in Kubernetes-CSI Prow jobs and Kubernetes itself: TODO
 
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind `CSIInlineVolume` feature flag
+- Initial unit tests and e2e tests implemented
+
+#### Beta
+
+- Feature flag enabled by default
+- CSI drivers can support both ephemeral inline volumes and persistent volumes
+
+#### GA
+
+- Remove dependency on deprecated `PodSecurityPolicy` and document new strategy
+- Upgrade / downgrade manual testing
+- Conformance tests implemented
```

Comment:
Would it be useful to leave the feature in beta and publish updated docs that don't rely on using PodSecurityPolicy, then get feedback for at least 1 release to make sure that folks are happy with how it works?

Reply:
I added a line to the GA graduation criteria to make sure the docs get updated before graduation. I'm not so sure it's necessary to wait another release cycle after the docs are updated since this feature has already been in beta for >2 years, but I'll defer that decision to the sig-storage leads.